documentation/development/steps_list.md

```diff
@@ -2,28 +2,55 @@
 
 _The following refers to all commands except `git-new-pull-request`, `git-repo` and `git-town`._
 
+The individual steps that each Git Town command executes are represested via the
+[command pattern](https://en.wikipedia.org/wiki/Command_pattern).
+This allows fully automated and robust `--continue`, `--abort`,
+and `--undo` functionality for each command.
+
+To distinguish the command-pattern commands from the Git Town commands in [src/cmd](../../src/cmd),
+we'll call the former `steps` from now on.
+
+
+## Running commands
+
 Each Git Town command begins by inspecting the current state of the Git repository
 (which branch you are on, whether you have open changes).
 If there are no errors, it generates a list of steps to run.
-Each step is a go `struct`.
 
-When executing a step, the undo steps are determined and added to a separate list.
-This is done by calling the methods `step.CreateUndoStepBeforeRun()` and `step.CreateUndoStepAfterRun()`.
+Steps, located in [src/steps](../../src/steps),
+implement the individual steps that each Git Town command performs,
+like for example [changing to a different Git branch](../../src/steps/checkout_branch_step.go)
+or [pulling down updates for the current branch](../../src/steps/pull_branch_step.go).
+They are Go structs that have a `Run` method which executes the step.
 
-## Abort / Continue
 
-If a Git command fails (typically due to a merge conflict), then the program halts
-and asks the user what they would like to do. In most cases they can either abort or continue.
+## Continuing commands
 
-If the user aborts then `step.CreateAbortStep()` is called and the undo steps are executed.
+If a Git command fails (typically due to a merge conflict),
+then the program halts and asks the user
+whether they would like to abort or continue.
```

Comment:
Thoughts on changing: 
`halts and asks the user whether they would like to abort or continue`
 to 
`saves state to disk, informs the user how to abort / continue, and exits`

I think thats more clear that the program flat out ends during this. We could change it so the program doesn't end and the user is actually given a prompt.

Reply:
I agree! done